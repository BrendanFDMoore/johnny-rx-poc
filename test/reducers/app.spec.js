import { REHYDRATE } from 'redux-persist/constants';

import reducers from 'reducers';
import * as Actions from 'actions';
import { ActionTypes } from 'constants/index';

describe('App', () => {
  it('should return the initial state', () => {
    expect(reducers.app(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${REHYDRATE} with payload`, () => {
    expect(reducers.app(undefined, { type: REHYDRATE, payload: { app: [] } })).toMatchSnapshot();
  });

  it(`should handle ${REHYDRATE} without payload`, () => {
    expect(reducers.app(undefined, { type: REHYDRATE, payload: { } })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.SHOW_ALERT}`, () => {
    expect(reducers.app(undefined, Actions.showAlert('success', 'hello')).notifications).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.HIDE_ALERT}`, () => {
    expect(reducers.app(undefined, Actions.hideAlert()).notifications).toMatchSnapshot();
  });
});
