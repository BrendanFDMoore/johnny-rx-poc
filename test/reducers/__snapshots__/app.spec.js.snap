exports[`App should handle HIDE_ALERT 1`] = `
Object {
  "message": "",
  "status": "",
  "visible": false,
  "withTimeout": true,
}
`;

exports[`App should handle SHOW_ALERT 1`] = `
Object {
  "message": "hello",
  "status": "success",
  "visible": true,
  "withTimeout": true,
}
`;

exports[`App should handle persist/REHYDRATE with payload 1`] = `
Object {
  "notifications": Object {
    "message": "",
    "status": "",
    "visible": false,
    "withTimeout": true,
  },
  "rehydrated": true,
}
`;

exports[`App should handle persist/REHYDRATE without payload 1`] = `
Object {
  "notifications": Object {
    "message": "",
    "status": "",
    "visible": false,
    "withTimeout": true,
  },
  "rehydrated": true,
}
`;

exports[`App should return the initial state 1`] = `
Object {
  "notifications": Object {
    "message": "",
    "status": "",
    "visible": false,
    "withTimeout": true,
  },
  "rehydrated": false,
}
`;
